### PR TITLE
Add bitbake tests and a few rpm tests. Also the -dev image has to be specified from now and on. 

### PR DESCRIPTION
**Pull request by fhajredini**
Bitbake tests help avoid regressions when working on bitbake. 

RPM tests that currently pass are also added to the test suite.

### CODE REVIEW

File: ci-scripts/yocto.groovy [hunk 1]
```diff
@@ -135,6 +135,23 @@ void runSmokeTests(String yoctoDir, String imageName) {
...
 }
 
+void runBitbakeTests(String yoctoDir) {
+    String archiveDir = "bitbake_TestReport"
+
+    stage("Perform Bitbake Testing"){
+        vagrant("/vagrant/cookbook/yocto/run-bitbake-tests.sh ${yoctoDir} ")
+    }
+    
+    //stage("Publish bitbake test results") {
```
Comment (sashko) [inline ci-scripts/yocto.groovy]: why do we need commented code?

**Comment by sashko** [comment]
did you run them on your machine with failing tests disabled?

File: ci-scripts/yocto.groovy (hunk 1 above)
Comment (fhajredini) [reply ci-scripts/yocto.groovy]: Im testing it. Not sure if publishing the test results is something specific of qemu tests or not.

**Comment by fhajredini** [comment]
> did you run them on your machine with failing tests disabled?

Not yet

**Comment by sashko** [comment]
@fhajredini why did you make the PR then?

**Comment by fhajredini** [comment]
> @fhajredini why did you make the PR then?

To test it on Jenkins.

**Comment by sashko** [comment]
> > @fhajredini why did you make the PR then?
> 
> To test it on Jenkins.

You can create your own build and not spam github.

**Comment by thenor** [comment]
The qemu test commit could have the domain in the beginning of the commit message short-text:
e.g. "smoke-tests: add RPM tests"

File: ci-scripts/yocto.groovy [hunk 2]
```diff
@@ -201,10 +232,18 @@ void buildManifest(String variantName, String imageName, String layerToReplace="
         // Build the images
         try {
             boolean buildUpdate = variantName.startsWith("rpi")
-            buildImageAndSDK(yoctoDir, imageName, variantName, buildUpdate)
             if (smokeTests) {
-                runSmokeTests(yoctoDir, imageName)
+                boolean weekly = env.WEEKLY_BUILD == "true"
+                buildDevImageForTesting(yoctoDir)
```
Comment (TSAnsari) [inline ci-scripts/yocto.groovy]: why not add the dev image in VARIANT_LIST of smoke test pipeline instead?

File: ci-scripts/yocto.groovy [hunk 3]
```diff
@@ -135,6 +135,15 @@ void runSmokeTests(String yoctoDir, String imageName) {
     }
 }
 
+void runBitbakeTests(String yoctoDir) {
+    String archiveDir = "bitbake_TestReport"
```
Comment (thenor) [inline ci-scripts/yocto.groovy]: You don't actually use this variable. Is it a remnant from the other version of this change?

File: ci-scripts/yocto.groovy [hunk 4]
```diff
@@ -82,6 +82,8 @@ void setupCache(String yoctoDir, String url) {
     }
 }
 
+
```
Comment (thenor) [inline ci-scripts/yocto.groovy]: This feels like an unnecessary edition of whitespace.

File: test-scripts/local.conf.appendix [hunk 5]
```diff
@@ -19,7 +19,6 @@ TEST_SUITES = " \
     systemd.SystemdBasicTests.test_systemd_failed \
     systemd.SystemdBasicTests.test_systemd_list \
     systemd.SystemdJournalTests.test_systemd_journal \
-    rpm.RpmBasicTest \
```
Comment (sashko) [inline test-scripts/local.conf.appendix]: why was this one removed?

**Pull request merged**